enterprise/frontend/src/embedding-sdk/cli/steps/pick-database-tables.ts

```diff
@@ -74,13 +74,23 @@ export const pickDatabaseTables: CliStepMethod = async state => {
   spinner.succeed();
 
   const chosenTables = await checkbox({
-    validate: choices => choices.length > 0,
-    message: "Select the tables to embed:",
+    validate: choices => {
+      if (choices.length === 0) {
+        return "Pick 1 - 3 tables to embed.";
+      }
+
+      if (choices.length > 3) {
+        return "You can only choose up to 3 tables.";
+      }
+
+      return true;
+    },
+    message: "Pick 1 - 3 tables to embed:",
     choices: tables.map(table => ({
       name: table.name,
       value: table,
     })),
   });
 
-  return [{ type: "done" }, { ...state, tables: chosenTables }];
+  return [{ type: "done" }, { ...state, tables, chosenTables }];
```

Comment:
We need to store both the found tables and the selected tables. We will later use the found tables to construct the permissions graph.